084-configure-env-vars-based-on-secrets-or-configmaps.md

```diff
@@ -0,0 +1,101 @@
+Strimzi itself will not directly extrapolate the content of the Secret or Config Map.
+It will configure the container environment variables to reference the Secret or Config Map and leave the extrapolation to Kubernetes.
+
+The enhanced `CointainerEnvVar` API would allow following options:
+
+```yaml
+    env:
+      - name: PLAIN_TEXT_ENV_VAR
+        value: "I'm plain text"
+      - name: SECRET_ENV_VAR
+        valueFrom:
+          secretKeyRef:
+            name: my-custom-secret
+            key: my-password
+      - name: CONFIG_ENV_VAR
+        valueFrom:
+          configMapKeyRef:
+            name: my-custom-cm
+            key: my-custom-option
+```
+
+In the same way as already done for the additional volumes, Strimzi will not validate the existence of the referenced Secret or Config Map.
+This validation will be done by Kubernetes when creating the Pod / Container.
+If the Secret or Config Map would not exist or would not contain the correct key, The Pod will not be started.
+This is considered acceptable given this is an advanced option.
+
+### Specifying environment variables based on other sources
+
+In addition to Secrets and Config Maps, Kubernetes also allows to set environment variables in containers based on other sources such as the Downward API.
+Strimzi will currently not support any additional sources other than Secrets and Config Maps.
+These sources can be added later based on a separate proposal.
+
+### Security risks
+
+The ability to use values from Secrets and Config Maps as environment variables is not validated against the RBAC rules of the user (or a service account) who creates the Strimzi custom resource.
+Following situation might occur and allow the user bypass the RBAC rights it has:
+1. Secret `my-secret` exists in a namespace watched by the Strimzi Cluster Operator
+2. User does not have the RBAC rights to read the Secret, but has the right to create Strimzi custom resources and exec into the Pods created by Strimzi
+3. The user creates any Strimzi custom resource and maps values from the Secret to environment variables
+4. Strimzi creates the Pod/container with the value from the Secret mapped to an environment variable
+5. User execs into the container created by Strimzi and checks the value of the environment variable
+
+This risk should be considered acceptable, mainly because:
+* This is standard Kubernetes behavior as the same could be done with any Kubernetes Workload API.
+* This can be already done while mounting the Secret or Config Map as a volume.
+* This risk already exists in Kafka Connect or MirrorMaker 2 through the external configuration.
+* The Secret or ConfigMap need to be in the same namespace os the Strimzi custom resource.
+  It would be rare that a user can create custom resources and exec into Pods in an namespace but cannot access some of its Secrets or Config Maps.
```

Comment:
```suggestion
This risk should be considered acceptable, mainly because:

* This is standard Kubernetes behavior as the same could be done with any Kubernetes Workload API.
* This can be already done while mounting the Secret or Config Map as a volume.
* This risk already exists in Kafka Connect or MirrorMaker 2 through the external configuration.
* The Secret or ConfigMap need to be in the same namespace of the Strimzi custom resource.
  It would be rare that a user can create custom resources and exec into Pods in an namespace but cannot access some of its Secrets or Config Maps.
```